crates/iota-sdk-ffi/src/crypto/secp256r1.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use iota_crypto::{Signer, Verifier};
+use iota_types::SignatureScheme;
+use rand::rngs::OsRng;
+
+use crate::{
+    error::{Result, SdkFfiError},
+    types::{
+        crypto::{Secp256r1PublicKey, Secp256r1Signature},
+        signature::{SimpleSignature, UserSignature},
+    },
+};
+
+#[derive(derive_more::From, uniffi::Object)]
+pub struct Secp256r1PrivateKey(pub iota_crypto::secp256r1::Secp256r1PrivateKey);
+
+#[uniffi::export]
+impl Secp256r1PrivateKey {
+    #[uniffi::constructor]
+    pub fn new(bytes: Vec<u8>) -> Result<Self> {
+        use iota_crypto::secp256r1::Secp256r1PrivateKey;
+        Ok(Self(Secp256r1PrivateKey::new(bytes.try_into().map_err(
+            |v: Vec<u8>| {
+                SdkFfiError::custom(format!("expected bytes of length 32, found {}", v.len()))
+            },
+        )?)))
+    }
+
+    pub fn scheme(&self) -> SignatureScheme {
+        self.0.scheme()
+    }
+
+    /// Get the public key corresponding to this private key.
+    pub fn public_key(&self) -> Secp256r1PublicKey {
+        self.0.public_key().into()
+    }
+
+    /// Sign a message and return a Secp256r1Signature.
+    pub fn try_sign(&self, message: &[u8]) -> Result<Secp256r1Signature> {
+        use iota_crypto::{Signer, secp256r1::Secp256r1PrivateKey};
+        use iota_types::Secp256r1Signature;
+        Ok(<Secp256r1PrivateKey as Signer<Secp256r1Signature>>::try_sign(&self.0, message)?.into())
```

Comment:
Can be simplified a bit
```suggestion
        use iota_crypto::Signer;
        use iota_types::Secp256r1Signature;
        Ok(Signer::<Secp256r1Signature>::try_sign(&self.0, message)?.into())
```
I guess decide between you and Thibault whether you want the import or the paths 🤷‍♀️

Reply:
The latest version you proposed in my PR is best IMO

Reply:
(without imports)

Reply:
https://github.com/iotaledger/iota-rust-sdk/pull/126/commits/5791a5689898d24dfd1cd650d3aaa67e093399b1
should be aligned with the Ed25519 PR now